src/components/sidebars/ScheduleSidebar.spec.jsx

```diff
@@ -0,0 +1,156 @@
+import { describe, it, expect, afterEach, vi } from "vitest";
+import { renderWithProvider } from "../../../test/utils/render-with-provider";
+
+import ScheduleSidebar from "./ScheduleSidebar";
+import {
+  suppliers,
+  contentfulSuppliers,
+} from "../../../test/fixtures/schedule-sidebar-state";
+import * as AppStatus from "../../constants/app-status";
+import { cleanup } from "@testing-library/react";
+import dateFormat from "dateformat";
+import { userEvent } from "@testing-library/user-event";
+import { scheduleAction } from "../../ContentfulWrapper";
+
+vi.mock("contentful-management", () => {
+  return {
+    createClient: vi.fn(),
+  };
+});
+
+vi.mock("@contentful/react-apps-toolkit", () => {
+  return {
+    useSDK: () => {
+      return {};
+    },
+  };
+});
+
+// we don't want to actually call Contentful in the tests
+vi.mock("../../ContentfulWrapper.js", () => {
+  return {
+    scheduleAction: vi.fn().mockResolvedValue({}),
+  };
+});
+
+const defaultComponentSetup = () => {
+  return renderWithProvider(<ScheduleSidebar />, {
+    preloadedState: {
+      suppliers: { value: suppliers },
+      contentfulSuppliers: { value: contentfulSuppliers },
+      appStatus: { value: AppStatus.PROCESSED_SUPPLIERS },
+    },
+  });
+};
+
+describe("ScheduleSidebar Component", () => {
+  afterEach(() => cleanup());
+
+  it("shows the number of suppliers that will be published", () => {
+    const { getByText } = defaultComponentSetup();
+    expect(
+      getByText(
```

Comment:
Learning/nosy question - in some specs you've been able to write things like `expect(getByText(todayString)).toBeTruthy();`; is there a reason these specs that check text/content have to include `(_, node) => node.textContent...`?

Reply:
Yes it's a bit annoying but if you pass a string into the `*ByText` methods, they don't match against text that is broken up across different HTML elements.  Because the `published` string is in a `<Badge>` component, the final HTML looks something like:

```
<li>
  3 suppliers will be
  <span>
     published
  </span>
</li>
```

So it doesn't match because it just looks element by element (or node by node).  If you pass a method in, you can write your own matcher with `node` - here I'm calling `.textContent` on it, so it will match against that.  If you call `textContent` on the `li` it will return the `published` text too, because it takes child nodes into account.